### PR TITLE
Super admin init and reset

### PR DESCRIPTION
**Pull request by d1str0**
Now allows for a 1 time only Super Admin initialization, with possible SA reset via command line.

POST /api/config/superadmin/{username}

Takes the given username, generates a random password, and returns the user object in JSON. Saves this user with super_admin role to the system.

How's it look, locke?

### CODE REVIEW

File: service/config.go [hunk 1]
```diff
@@ -32,10 +45,52 @@ func (s *configService) SetJWTSecret(secret []byte) error {
...
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.userService.SaveUser(u)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.repo.SetSuperAdmin(name)
```
Comment (brooks42) [inline service/config.go:75]: Question might just be cuz I don't quite understand the backend arch yet but if save user fails with an error are we stuck in a place where we made a user but haven't set super admin permissions to that user?

File: service/config.go [hunk 2]
```diff
@@ -32,10 +45,52 @@ func (s *configService) SetJWTSecret(secret []byte) error {
...
+	return u, p, err
+}
+
+// ResetSuperAdmin will delete the currently stored username for Super Admin,
+// which allows a new Super Admin to be initialized.
+func (s *configService) ResetSuperAdmin() error {
+	return s.repo.SetSuperAdmin("")
+}
+
+// newPassword generates 32 cryptographically secure random bytes, base64
```
Comment (brooks42) [inline service/config.go:85]: if we ever up this to 64 bits this comment is gonna be wrong, maybe removing the specifics would be a good idea?

**Review by brooks42 — APPROVED** [review]
Just some questions and comments but nothing blocking

File: service/config.go (hunk 1 above)
Comment (d1str0) [reply service/config.go:75]: If SaveUser fails, the user is just never created really. Although if SetSuperAdmin fails, the User will still exist and have been saved so I should fix that.

File: service/config.go (hunk 2 above)
Comment (d1str0) [reply service/config.go:85]: yeah probably

**Pull request merged**